src/commands/UndeployVDBCommand.ts

```diff
@@ -14,17 +14,41 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import * as vscode from 'vscode';
 import * as extension from '../extension';
 import { DVProjectTreeNode } from '../model/tree/DVProjectTreeNode';
+import * as kubectlapi from 'vscode-kubernetes-tools-api';
+import * as utils from '../utils';
+import * as path from 'path';
+import { handleReferences } from './DeployVDBCommand';
 
-export function undeployVDBCommand(ctx) {
-	if (ctx instanceof DVProjectTreeNode) {
-		const prjNode: DVProjectTreeNode = ctx;
-		const file: string = prjNode.file;
-		handleUndeploy(file);
+export async function undeployCommand(prjNode: DVProjectTreeNode) {
+	await handleReferences(prjNode, false);
+	const alreadyDeployed: boolean = await utils.isVDBDeployed(prjNode.dvConfig.metadata.name);
+	if (alreadyDeployed) {
+		await handleUndeploy(prjNode.dvConfig.metadata.name, path.basename(prjNode.dvConfig.metadata.name, '.yaml'));
 	}
 }
 
-function handleUndeploy(filepath: string): void {
-	extension.log(`UNDEPLOY: Selected File: ${filepath}`);
+async function handleUndeploy(file: string, name: string, suppressOutputs: boolean = false): Promise<void> {
+	const k8sApi: kubectlapi.API<kubectlapi.KubectlV1> = await kubectlapi.extension.kubectl.v1;
+	if (k8sApi && k8sApi.available) {
+		try {
+			extension.log(`Deleting resource ${name}...`);
```

Comment:
I think we need to be more precise. We are "deleting" on the remote instance. In the Ui, we are even usin gthe terminology Undeploy.
Just deleting without precising more, I was thinking that my local file was deleted

Reply:
Yeah, fair point.